### ISSUE TITLE
Open class files in VSCode and add a debug breakpoint anywhere

### ISSUE THREAD

**Issue by ydmykr** [redhat-developer/vscode-java#2564]
I'm trying to add a breakpoint in a class library. I must first open the target library class file (decompiled), and add a breakpoint later.

![image](https://user-images.githubusercontent.com/100815370/178924629-eb0dd836-69aa-4d1e-b3a2-18cbe950b14a.png)

But it looks like I can view the class library only while debugging the java application. How can I goto specific library class file in vscode?

**Comment by ydmykr** [comment]
@[testforstephen](https://github.com/testforstephen) Does it work for symbols outside the project? I'm trying to set a breakpoint on spring framework, which is a library imported from maven.

**Comment by testforstephen** [comment]
yes, it searches for all classes including source files and library classes.

**Comment by CaledoniaProject** [comment]
@testforstephen It works for class names, not class methods, is that expected?

![screen 2022-07-15 at 13 50 18](https://user-images.githubusercontent.com/1357701/179159546-43b4991e-f02a-400b-bcec-e7f2f97f5356.jpg)

**Comment by testforstephen** [comment]
Yes, it only searchs the classes by default, that's by design.

We have a setting `java.symbols.includeSourceMethodDeclarations` to control whether to include methods from source files during symbol search. This is off by default. However, it does not include the methods in the library due to performance considerations.

Usually, I use `Ctrl + T` to search the target class first, then using `Ctrl + Shift + O` to search method in current opened file.

**Comment by CaledoniaProject** [comment]
> However, it does not include the methods in the library due to performance considerations.

Any chances to create a new setting for that?

**Comment by testforstephen** [comment]
No plan yet. It's very slow to search methods from all libraries.

**Comment by gayanper** [comment]
I think with new parallel index search in jdt.core searching for methods should be fast. I have a eclipse implementation as a plugin just for this which is faster for larger workspaces as well. So may be having setting might be good to use as a circuit breaker for really slow workspaces.
